src/common/k8s-api/kube-object.store.ts

```diff
@@ -456,30 +457,46 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
   protected updateFromEventsBuffer() {
     const items = this.getItems();
 
-    for (const { type, object } of this.eventsBuffer.clear()) {
-      const index = items.findIndex(item => item.getId() === object.metadata?.uid);
-      const item = items[index];
+    for (const event of this.eventsBuffer.clear()) {
+      if (event.type === "ERROR") {
+        continue;
+      }
 
-      switch (type) {
-        case "ADDED":
+      try {
+        const { type, object } = event;
 
-          // falls through
-        case "MODIFIED": {
-          const newItem = new this.api.objectConstructor(object);
+        if (!object.metadata?.uid) {
```

Comment:
How/when it can happen? _(existing object without `ID`)_

Reply:
I don't know just saw this exception happening from sentry in 5.4. The exception was that object was undefined so that reading metadata threw a type error.

Reply:
Then condition should be `if (!object?.metadata.uid)`